### PR TITLE
Issue#655 support for building with Clang

### PR DESCRIPTION
**Pull request by monadbobo**
Compiled by clang 7.0.1((Fedora 7.0.1-6.fc29)

### CODE REVIEW

**Comment by monadbobo** [comment]
jenkins go!

**Comment by monadbobo** [comment]
close issue #655 .

File: src/interface/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,10 @@
-add_compile_options(-Wno-maybe-uninitialized)
+if (CMAKE_GUN_COMPILER_ID)
+  add_compile_options(-Wno-maybe-uninitialized)
+endif()
```
Comment (sherman-the-tank) [inline src/interface/CMakeLists.txt]:
```
elseif(...)
```
??

File: CMakeLists.txt [hunk 2]
```diff
@@ -300,7 +300,15 @@ add_compile_options(-Wunused-parameter)
 add_compile_options(-Wshadow)
 
 # This requries GCC 5.1+
-add_compile_options(-Wsuggest-override)
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 5.1)
+   add_compile_options(-Wsuggest-override)
+endif()
```
[the remark this answers is not in the document]
Comment (monadbobo) [reply CMakeLists.txt]: Thanks, I will refactor this .

File: src/common/base/test/EitherOrTest.cpp [hunk 3]
```diff
@@ -362,15 +362,17 @@ TEST(EitherOr, MoveOutValue) {
 TEST(EitherOr, SelfAssignment) {
     {
         EitherOr<int, std::string> r("SomeValue");
-        r = r;
-        ASSERT_TRUE(r.isRightType());
-        EXPECT_EQ("SomeValue", r.right());
+        decltype(r) r1;
```
[the remark this answers is not in the document]
Comment (monadbobo) [reply src/common/base/test/EitherOrTest.cpp]: Ok, I will fix this.

File: src/interface/CMakeLists.txt (hunk 1 above)
Comment (monadbobo) [reply src/interface/CMakeLists.txt]: This was a mistake.

**Pull request merged**